### DISCUSSION TITLE
Interval based Persist functionality

### DISCUSSION THREAD

**Discussion by YashChauhan858** [pmndrs/zustand#2631] [Ideas]
I'm working on a crypto based project where i need to show user price as soon as they comes on the page and not wait for price to fetch one by one ( real time ) for each crypto pair ( BTC/USD, ETH/USD ), for this to work i made a react based hook that does the same thing but I'm thing it would be better if there is an option that let us  do so in persist middelware.

**Comment by dai-shi** [comment]
Hi, thanks for opening up a discussion.
I wonder if you can use the `storage` option and provide your own storage.
If it doesn't work, another option would be to create a third party library in https://github.com/zustandjs.

**Reply by YashChauhan858** [reply]
OK i created a third party package https://www.npmjs.com/package/zustand-interval-persist for this feature
how can i get it included on https://github.com/pmndrs/zustand/blob/main/docs/integrations/third-party-libraries.md page

this is my first time writing a package , so if you can guide me with some good practice and what to do to get it included would be great

thanks 😊

**Reply by dai-shi** [reply]
Hi, you can open a PR to modify third-party-libraries.md.

Currently, it's a wrapper function around `create`.
If you want to improve it, you might want to consider providing a middleware, so that it allows users to use it with other middleware such as `devtools`.
However, it might be very challenging to make correct types in TypeScript.
